sources/nodes/coreos.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nodes
+
+import (
+	"flag"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	fleetClient "github.com/coreos/fleet/client"
+	"github.com/coreos/fleet/etcd"
+	fleetPkg "github.com/coreos/fleet/pkg"
+	"github.com/coreos/fleet/registry"
+	"github.com/golang/glog"
+)
+
+var argFleetEndpoints = flag.String("fleet_endpoints", "http://127.0.0.1:4001", "Comma separated list of fleet server endpoints")
+
+const etcdRegistry = "/_coreos.com/fleet/"
+
+type fleetNodes struct {
+	client         fleetClient.API
+	nodes          *NodeList
+	apiErrors      int
+	recentApiError error
+}
+
+func (self *fleetNodes) List() (*NodeList, error) {
+	nodes, err := self.client.Machines()
+	if err != nil {
+		self.apiErrors++
+		self.recentApiError = err
+		glog.V(1).Infof("failed to get list of machines from fleet - %q", err)
+		return nil, err
+	}
+	nodeList := newNodeList()
+	for _, node := range nodes {
+		nodeList.Items[Host(node.ID)] = Info{PublicIP: node.PublicIP, InternalIP: node.PublicIP}
+	}
+	self.nodes = nodeList
+	return nodeList, nil
+}
+
+func (self *fleetNodes) DebugInfo() string {
+	output := fmt.Sprintf("Fleet Nodes plugin: Aggregate error count: %d; recent error: %v", self.apiErrors, self.recentApiError)
+	if self.nodes != nil {
+		output = fmt.Sprintf("%s\nCadvisor Nodes: %v", output, self.nodes.Items)
+	}
+	return output
+}
+
+func getFleetRegistryClient(fleetEndpoints []string) (fleetClient.API, error) {
+	var dial func(string, string) (net.Conn, error)
+
+	tlsConfig, err := fleetPkg.ReadTLSConfigFiles("", "", "")
```

Comment:
This won't let us do HTTPS no? What is the alternative to this? Do you actually have to give it the file paths?